front/lib/api/assistant/actions/retrieval.ts

```
@@ -190,107 +183,22 @@ export async function retrievalActionSpecification(
+    actionConfiguration,
+  }: {
+    actionConfiguration: RetrievalConfigurationType;
   }
-
-  const spec = await retrievalActionSpecification(actionConfig);
-
-  if (spec.inputs.length > 0) {
-    const now = Date.now();
-
-    const rawInputsRes = await generateActionInputs(
-      auth,
-      configuration,
-      spec,
-      conversation,
-      userMessage
-    );
-
-    if (rawInputsRes.isOk()) {
-      const rawInputs = rawInputsRes.value;
-
-      logger.info(
-        {
-          workspaceId: conversation.owner.sId,
-          conversationId: conversation.sId,
-          elapsed: Date.now() - now,
-        },
-        "[ASSISTANT_TRACE] Retrieval action inputs generation"
-      );
-
-      if (actionConfig.query === "auto") {
-        if (!rawInputs.query || typeof rawInputs.query !== "string") {
-          return new Err(
-            new Error("Failed to generate a valid retrieval query.")
-          );
-        }
-        query = rawInputs.query as string;
-      }
-
-      if (actionConfig.relativeTimeFrame === "auto") {
-        if (
-          rawInputs.relativeTimeFrame &&
-          typeof rawInputs.relativeTimeFrame === "string"
-        ) {
-          relativeTimeFrame = parseTimeFrame(rawInputs.relativeTimeFrame);
-        }
-      }
-    } else {
-      logger.error(
-        {
-          workspaceId: conversation.owner.sId,
-          conversationId: conversation.sId,
-          elapsed: Date.now() - now,
-          error: rawInputsRes.error,
-        },
-        "Error generating retrieval action inputs"
-      );
-
-      // We fail the rerieval only if we had to generate a query but failed to do so, if the
-      // relativeTimeFrame failed, we'll just use `null`.
-      if (actionConfig.query === "auto") {
-        return rawInputsRes;
-      }
-    }
+): Promise<Result<AgentActionSpecification, Error>> {
+  const owner = auth.workspace();
+  if (!owner) {
+    throw new Error("Unexpected unauthenticated call to `runRetrieval`");
```

Why not return an Error since this function returns a Result?

---

Here if we don't have an owner it's a 500, no point in returning an error IMO

---

Yep this is a perfect case for 500 there is no world in which we would act specifically on this error in any meaningful way other than a 500

---

Ok, I guess it's an issue with our auth object then, cause IMHO we should not have to have so many of those checks everywhere. Only the front-line functions should.

---

could have sub-types where owner is not optional